### PR TITLE
feat: search component #647

### PR DESCRIPTION
**Pull request by MS151994**

### CODE REVIEW

File: app/components/SearchMoviesInput/SearchMoviesInput.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  const placeholder = {
+    searchValue: 'Title, Actors, Film genre',
+  };
+  const inputRef = createRef();
+
+  const closeIcon = <>&times;</>;
+  const SearchIcon = <FontAwesomeIcon icon={faSearch} />;
+  const containerWidth = isExpanded ? 300 : 30;
+  const containerBackground = isExpanded ? 'rgba(0, 0, 0, 0.8)' : 'none';
```
Comment (wlechowicz) [inline app/components/SearchMoviesInput/SearchMoviesInput.js:23]: These styles shouldn't be defined here. You can pass the isExpanded prop and use in the CSS to change the styles by defining the collapsed state as default, then like this
```
width: 30px;
background...

${({ isExpanded }) => isExpanded && css(`
  width: 300px;
  background...
`)}

**Review by wlechowicz — CHANGES_REQUESTED** [review]
Please rename 
- app/components/SearchMoviesInput/SearchMoviesInput.js -> app/components/SearchMoviesInput/index.js
- app/components/SearchMoviesInput/searchMovieInputStyles.js -> app/components/SearchMoviesInput/SearchMoviesInput.styles.js

**Review by wlechowicz — CHANGES_REQUESTED** [review]

File: app/components/SearchMoviesInput/SearchMoviesInput.styles.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import styled, { css } from 'styled-components';
+import { device } from '/app/consts/mediaQueries';
```
Comment (wlechowicz) [inline app/components/SearchMoviesInput/SearchMoviesInput.styles.js:2]:
```suggestion
import { device } from 'consts/mediaQueries';
```

**Review by wlechowicz — CHANGES_REQUESTED** [review]

File: app/components/SearchMoviesInput/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    <SearchMoviesContainer isExpanded={isExpanded} ref={ref}>
+      <SearchButton onClick={openSearchInput}>{SearchIcon}</SearchButton>
+      <SearchInput
+        placeholder={placeholder.searchValue}
+        value={searchInputValue}
+        onChange={(e) => setSearchInputValue(e.target.value)}
+        ref={inputRef}
+      ></SearchInput>
+      {searchInputValue.length > 0 && (
```
[the remark this answers is not in the document]
Comment (MS151994) [reply app/components/SearchMoviesInput/index.js:44]: like this? 
`const [isInputEmpty, setInputEmpty] = useState(true);
 
  const changeInputValue = (e) => {
    const value = e.target.value;
    setSearchInputValue(value);
    value.length === 0 ? setInputEmpty(true) : setInputEmpty(false);
  };`
  
  onChange={(e) => changeInputValue(e)}

**Comment by rast-mk1** [comment]
Is Search Component supposed to be invisible on mobile? 

obraz

**Comment by MS151994** [comment]
> Is Search Component supposed to be also in mobile?
> 
> obraz

i don't know. i suggested original netflix site on web browser, there is no search there component on mobile.

File: app/components/SearchMoviesInput/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (wlechowicz) [reply app/components/SearchMoviesInput/index.js]: No, just `const isInputEmpty = searchInputValue.length > 0;`

File: app/components/SearchMoviesInput/index.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  const containerRef = useRef();
+
+  useOnClickOutside(containerRef, () => (!isInputEmpty ? setExpanded(true) : setExpanded(false)));
+
+  const openSearchInput = () => {
+    setExpanded(!isExpanded);
+    if (!isExpanded) {
+      inputRef.current.focus();
+    }
+  };
```
Comment (marcelina-hasiak) [inline app/components/SearchMoviesInput/index.js:34]: I can have some value inside input and can't close it when I click outside, but I can close it when I click SearchButton - missing logic for checking if input is empty

**Review by marcelina-hasiak — CHANGES_REQUESTED** [review]

**Comment by rast-mk1** [comment]
What do you think about writing story or stories for `SearchMoviesInput` so there's Storybook preview?

File: app/components/SearchMoviesInput/SearchMoviesInput.styles.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  height: 30px;
+  margin-right: 5px;
+  cursor: pointer;
+  border: none;
+  font-size: 1.1rem;
+  color: #fff;
+`;
+
+export const SearchIconContainer = styled(SearchButton)``;
+
```
Comment (hiimsagan) [inline app/components/SearchMoviesInput/SearchMoviesInput.styles.js:41]: Are you sure you wanna expand button? Mabye Some div ?

Comment (marcelina-hasiak) [reply app/components/SearchMoviesInput/SearchMoviesInput.styles.js:41]: This component is unnecessary.

**Review by marcelina-hasiak — CHANGES_REQUESTED** [review]
Too much to talk about here, I'll leave a piece of advice advices in DM

**Pull request closed without merging**